### PR TITLE
hoc2023: AI dancers

### PR DESCRIPTION
**Pull request by breville**
This adds support for optional background dancers in the AI block, which is being added in https://github.com/code-dot-org/code-dot-org/pull/53983.

### CODE REVIEW

File: src/p5.dance.js [hunk 1]
```diff
@@ -184,6 +184,11 @@ module.exports = class DanceParty {
   }
 
   async loadCostumeAnimations(costume, costumeData) {
+    if (!this.animations[costume]) {
+      // Invalid costume, nothing to do:
```
Comment (fisher-alice) [inline src/p5.dance.js:188]: Add a console statement similar to line 974.

File: src/p5.dance.js [hunk 2]
```diff
@@ -966,7 +971,7 @@ module.exports = class DanceParty {
         sprite.rotation = 0;
       });
     } else {
-      throw new Error('Unexpected format: ' + format);
+      console.log('Unexpected layout format: ' + format);
```
Comment (fisher-alice) [inline src/p5.dance.js:974]: Just noting that if OpenAI responds with an invalid costume, we ignore and will not display dancers. However, the generated code and explanation will still reference the hallucinogenic response, in the example below, the costume 'rabbit'.
Screen Shot 2023-10-02 at 3 26 10 PM

**Review by fisher-alice — APPROVED** [review]
Nice addition!

File: src/p5.dance.js (hunk 2 above)
Comment (breville) [reply src/p5.dance.js:975]: Yes, it's a good point.  I think we should work out a plan for detecting this situation and explaining it as a teachable moment!

File: src/p5.dance.js (hunk 1 above)
Comment (breville) [reply src/p5.dance.js:189]: Done.

**Pull request merged**